### PR TITLE
Add function to add custom domain and verify it

### LINKED ISSUE #45: Domain Verification for Custom Domains
**Description:** This issue involves implementing domain verification methods to ensure that users own the custom domains they intend to use. Domain verification is crucial for security and authorization purposes.
- **Tasks:**
  - Define and implement domain verification mechanisms (e.g., DNS TXT records, HTML file upload, meta tags).
  - Document the domain verification process.
  - Ensure that only verified domains can be used for custom short URLs.
- **Acceptance Criteria:** Users should be able to verify ownership of their custom domains, and the system should restrict the use of unverified domains.
- **Additional Information:** Detailed instructions on domain verification can be found in our [Custom Domain Feature Implementation Guide](https://github.com/DhananjayThomble/URL-Shortener-App/wiki/Custom-Domain-Feature-Implementation-Guide).

### PR DESCRIPTION
**Pull request by ArunBohra12**
## What does this PR do?
Provide function/routes to add custom domain for a user. Also adds endpoint to verify their domain

Fixes #45 

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How should this be tested?

Currently manual

- Are there environment variables that should be set? - No
- What are the minimal test data to have? - A DNS record at minimum

### CODE REVIEW

File: backend/controllers/domainControllers/verifyCustomDomain.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      console.error(`Invalid data present for the user ${user._id}`);
+
+      return res.status(200).json({
+        message: 'Sorry, something went wrong! Please try again',
+      });
+    }
+
+    const domainVerification = await checkDNSTxtVerification(
+      // This URL might need change
+      `_verification.${user.customDomain.url}`,
```
Comment (DhananjayThomble) [inline backend/controllers/domainControllers/verifyCustomDomain.js:31]: Remove the `_verification.` from the 1st parameter. here we can directly pass the domain name in the first parameter and the verification code (for lookup) in the second parameter.

**Review by DhananjayThomble — CHANGES_REQUESTED** [review]
Hello @ArunBohra12, pls review the requested changes and commit them accordingly.

File: backend/controllers/domainControllers/verifyCustomDomain.js (hunk 1 above)
Comment (ArunBohra12) [reply backend/controllers/domainControllers/verifyCustomDomain.js]: Removed the `_verification` part from the domain.

**Comment by ArunBohra12** [comment]
Still need to work on checking continuously for verification.

**Review by DhananjayThomble — APPROVED** [review]

**Comment by ArunBohra12** [comment]
Decided to go with cron job method, currently it is ok but with the scaling we'll need some other solution.

**Review by DhananjayThomble — APPROVED** [review]

**Pull request merged**